.github/workflows/docker.yml

```diff
@@ -0,0 +1,95 @@
+name: Build and publish docker images
+on:
+  workflow_call:
+    inputs:
+      arm64:
+        description: If set to true, build an image for arm64 as well as amd64.
+        type: boolean
+        required: false
+        default: false
+      image-name:
+        description: "What to call the docker image. This is the FOO in 'matrixdotorg/FOO:tagname'"
```

Comment:
I would choose to make the end-user include `matrixdotorg/` in the name they specify.

For two reasons really;
* searchability in the repo that uses this action; and mainly
* when prototyping, it's helpful to publish to a personal namespace.

Reply:
Agreed, thanks